crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -1538,12 +1537,15 @@ mod tests {
         assert_matches!(rx.await, Ok(Ok(())));
     }
 
-    fn insert_blocks<'a, DB: Database>(db: &DB, mut blocks: impl Iterator<Item = &'a SealedBlock>) {
-        let mut transaction = Transaction::new(db).unwrap();
-        blocks
-            .try_for_each(|b| transaction.insert_block(b.clone(), None))
-            .expect("failed to insert");
-        transaction.commit().unwrap();
+    fn insert_blocks<'a, DB: Database>(
+        db: &DB,
+        chain: Arc<ChainSpec>,
+        mut blocks: impl Iterator<Item = &'a SealedBlock>,
+    ) {
+        let factory = ShareableDatabase::new(db, chain);
+        let mut provider = factory.provider_rw().unwrap();
```

Comment:
when can this panic?

Reply:
when it fails to open the inner db transaction, similar to the previous one